odoo/tools/query.py

```diff
@@ -87,15 +79,20 @@ def make_alias(self, alias: str, link: str) -> str:
     def add_table(self, alias: str, table: (str | SQL | None) = None):
         """ Add a table with a given alias to the from clause. """
         assert alias not in self._tables and alias not in self._joins, f"Alias {alias!r} already in {self}"
-        self._tables[alias] = _sql_table(table)
+        table = table or alias
+        if isinstance(table, str):
+            table = SQL.identifier(table)
+        self._tables[alias] = table
```

Comment:
I think we could enforce table's type to be `SQL` or `None`, just like in the constructor API.